doc/man3/RAND_DRBG_get0_master.pod

```diff
@@ -0,0 +1,81 @@
+=pod
+
+=head1 NAME
+
+RAND_DRBG_get0_master,
+RAND_DRBG_get0_public,
+RAND_DRBG_get0_private
+- get access to the global RAND_DRBG instances
+
+=head1 SYNOPSIS
+
+ #include <openssl/rand_drbg.h>
+
+ RAND_DRBG *RAND_DRBG_get0_master(void);
+ RAND_DRBG *RAND_DRBG_get0_public(void);
+ RAND_DRBG *RAND_DRBG_get0_private(void);
+
+
+=head1 DESCRIPTION
+
+These functions are part of the OpenSSL L<RAND_DRBG(7)> API.
+
+The default RAND API implementation (RAND_OpenSSL()) utilizes three
+shared DRBG instances which are accessed via the RAND API:
+
+The <public> and <private> DRBG are thread-local instances, which are used
+by RAND_bytes() and RAND_priv_bytes(), respectively.
+The <master> DRBG is a global instance, which is not intended to be used
+directly, but is used internally to reseed the other two instances.
+
+These functions here provide access to the shared DRBG instances.
+
+=head1 RETURN VALUES
+
+RAND_DRBG_get0_master() returns a pointer to the <master> DRBG instance.
+
+RAND_DRBG_get0_public() returns a pointer to the <public> DRBG instance.
+
+RAND_DRBG_get0_private() returns a pointer to the <private> DRBG instance.
+
+
+=head1 NOTES
+
+It is not thread-safe to access the <master> DRBG instance.
+The <public> and <private> DRBG instance can be accessed safely, because
```

Comment:
I'd include something to indicate that the returned pointers should not be stored if a switch to another thread is possible.

I'm unsure exposing these is wise because of the possibility of confusion.  I've no trouble with get_bytes or get_priv_bytes being exposed since they will always choose the current thread's drbg.

Reply:
> I'd include something to indicate that the returned pointers should not be stored if a switch to another thread is possible.

Thanks for the suggesion, I will add an improvement.

> I'm unsure exposing these is wise because of the possibility of confusion. I've no trouble with get_bytes or get_priv_bytes being exposed since they will always choose the current thread's drbg.

Using the per-thread DRBGs is safe as long as one obeys the rules. For example, I could create per-object DRBG instances which are chained to one of the per-thread DRBGs if I know that my objects are owned by a specific worker thread.

In RAND_DRBG(7) I placed a disclaimer that this API is not intended for a typical user needing random data and that the RAND API should be used instead.  It's more a "backend" API for people (like me) who want to (or have to) fiddle with the entropy sources. And we can assume those kind of people know what they are doing. ;-)